File: backend/src/main/java/com/team01/backend/domain/user/dto/UserResponseDto.java
package com.team01.backend.domain.user.dto;

import com.team01.backend.domain.user.entity.Role;
import com.team01.backend.domain.user.entity.User;

public record UserResponseDto(
        String email,
        String nickname,
        String profileImage,
        Role role
) {
    public UserResponseDto(User user) {
        this(
            user.getEmail(),
            user.getNickname(),
            user.getProfileImage(),
            user.getRole()
        );
    }
}

File: backend/src/main/java/com/team01/backend/domain/user/dto/UserResponseDto.kt
package com.team01.backend.domain.user.dto

import com.team01.backend.domain.user.entity.Role
import com.team01.backend.domain.user.entity.User

data class UserResponseDto(
    val email: String,
    val nickname: String,
    val profileImage: String?,
    val role: Role
) {
    constructor(user: User) : this(
        user.email,
        user.nickname,
        user.profileImage,
        user.role
    )
}
